README.md

```diff
@@ -67,3 +67,48 @@ jobs:
           chmod +x dist/bin/*
           dist/bin/${{ matrix.bin }} --version
 ```
+
+## Workflows
+
+### [`haskell-ci`](./.github/workflows/haskell-ci.yml)
+
+Build and test single-package Haskell projects using Cabal. Performs a sequence
+of steps that are appropriate for many projects, namely:
+
+- Uses `haskell-actions/setup` to install Cabal and GHC
+- Runs `cabal sdist`, unpack the `sdist`, and perform subsequent steps there
+- Uses `actions/cache` to cache the Cabal store and build directory (`dist-newstyle`)
+- Runs `cabal configure`, `build`, `test`, `sdist`, `check`, and `haddock`
+- Unconditionally saves the cache
+- Checks that the package is compatible with GHC's bundled version of Cabal
+
+This workflow will not be applicable to all projects, but it is applicable to
+enough to be useful.
+
+For a description of the inputs, see the workflow itself.
```

Comment:
There is quite a bit of overlap in functionality between this reusable workflow and the goals of the [`haskell-ci`](https://github.com/haskell-CI/haskell-ci) tool (not to be confused with the `haskell-ci` workflow you've defined here). I think it would still be worthwhile to have both, but I wonder if this README should include a brief compare/contrast with the `haskell-ci` tool.

Reply:
Added an initial comparison, please let me know if you have further thoughts. I'd bet that you're more familiar with haskell-ci (the tool) than I am.

Reply:
This looks great!

I think the most prominent difference in how each of these work is that this reusable workflow is meant to be imported in an existing CI YAML configuration, whereas the `haskell-ci` tool generates a standalone YAML configuration. This means that the latter will generally produce noisier diffs when updating, as it may require recreating large chunks of autogenerated YAML.

One of the latter's strengths is that is has various extra features that the former does not currently have, such as automatic integration of tools like `hlint` and `docspec`. We might want to say that we consider such features out of scope for the former, as we prefer to handle this in separate workflows.

Reply:
Oh, hah, I forgot to mention their biggest and most important differences :smile: 

Added, thanks!